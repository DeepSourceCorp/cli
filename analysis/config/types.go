package config

import "github.com/deepsourcelabs/cli/analysis/lsp"

///////////////////////////////////////////////////////
// DSConfig is the struct for .deepsource.toml file //
/////////////////////////////////////////////////////

type DSConfig struct {
	Version         int           `mapstructure:"version,omitempty" json:"version" toml:"version"`
	ExcludePatterns []string      `mapstructure:"exclude_patterns,omitempty" json:"exclude_patterns,omitempty" toml:"exclude_patterns"`
	TestPatterns    []string      `mapstructure:"test_patterns,omitempty" json:"test_patterns,omitempty" toml:"test_patterns"`
	Analyzers       []Analyzer    `mapstructure:"analyzers,omitempty" json:"analyzers,omitempty" toml:"analyzers"`
	Transformers    []Transformer `mapstructure:"transformers,omitempty" json:"transformers,omitempty" toml:"transformers"`
}

type Analyzer struct {
	Name                string      `mapstructure:"name,omitempty" json:"name,omitempty" toml:"name"`
	RuntimeVersion      string      `mapstructure:"runtime_version,omitempty" json:"runtime_version,omitempty" toml:"runtime_version,omitempty"`
	Enabled             bool        `mapstructure:"enabled,omitempty" json:"enabled" toml:"enabled"`
	DependencyFilePaths []string    `mapstructure:"dependency_file_paths,omitempty" json:"dependency_file_paths,omitempty"`
	Meta                interface{} `mapstructure:"meta,omitempty" json:"meta,omitempty" toml:"meta"`
	Thresholds          interface{} `mapstructure:"thresholds,omitempty" json:"thresholds,omitempty"`
}

type Transformer struct {
	Name    string `mapstructure:"name,omitempty" json:"name,omitempty" toml:"name"`
	Enabled bool   `mapstructure:"enabled,omitempty" json:"enabled,omitempty" toml:"enabled"`
}

///////////////////////////////////////
// LSP based Analysis Config Types  //
/////////////////////////////////////

type AnalysisConfig struct {
	Files           []lsp.TextDocumentItem `json:"files"`
	TestFiles       []lsp.TextDocumentItem `json:"test_files"`
	ExcludedFiles   []lsp.TextDocumentItem `json:"excluded_files"`
	ExcludePatterns []string               `json:"exclude_patterns"`
	TestPatterns    []string               `json:"test_patterns"`
	AnalyzerMeta    interface{}            `json:"analyzer_meta"`
}

//////////////////////////////////////
// LSP based Analysis Result Types  //
/////////////////////////////////////

type Namespace struct {
	Key   string  `json:"key"`
	Value float64 `json:"value"`
}

type Metric struct {
	MetricCode string      `json:"metric_code"`
	Namespaces []Namespace `json:"namespaces"`
}

type AnalysisResult struct {
	Issues   []lsp.Diagnostic `json:"issues"`
	Metrics  []Metric         `json:"metrics,omitempty"`
	IsPassed bool             `json:"is_passed"`
	Errors   []Error          `json:"errors"`
	// Errors    []lsp.Diagnostic `json:"errors"`
	ExtraData interface{} `json:"extra_data"`
}

////////////////////////////////////
// Default Analysis Result Types  //
///////////////////////////////////

type Error struct {
	HMessage string `json:"hmessage"`
	Level    int    `json:"level"`
}

type Coordinate struct {
	Line   int `json:"line"`
	Column int `json:"column"`
}

type Position struct {
	Begin Coordinate `json:"begin"`
	End   Coordinate `json:"end"`
}

type Location struct {
	Path     string   `json:"path"`
	Position Position `json:"position"`
}

type Issue struct {
	Code     string   `json:"issue_code"`
	Title    string   `json:"issue_text"`
	Location Location `json:"location"`
}

type DefaultAnalysisResult struct {
	Issues    []Issue     `json:"issues"`
	Metrics   []Metric    `json:"metrics,omitempty"`
	IsPassed  bool        `json:"is_passed"`
	Errors    []Error     `json:"errors"`
	ExtraData interface{} `json:"extra_data"`
}
