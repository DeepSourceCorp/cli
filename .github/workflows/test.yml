name: Unit Tests

on:
  push:
    branches: [develop]
  pull_request:
    branches: [develop]

jobs:
  test:
    runs-on: ubuntu-latest
    env:
      NODE_AUTH_TOKEN: ${{secrets.GITHUB_TOKEN}}
      ZEAL_SECRET: ${{secrets.GITHUB_TOKEN}}

    steps:
      - name: Checkout Bifrost
        uses: actions/checkout@v2

      - name: Setting Up Node
        uses: actions/setup-node@v2
        with:
          node-version: '14.x'
          always-auth: true
          registry-url: https://npm.pkg.github.com/
          scope: '@deepsourcelabs'
          cache: 'yarn'

      - name: Install packages
        run: yarn

      - name: Run Test
        run: yarn test --coverage=true --coverageReporters=cobertura

      - name: Install DeepSource CLI
        run: curl https://deepsource.io/cli | sh

      - name: Set DEEPSOURCE_DSN
        run: |
          ./bin/deepsource report --analyzer test-coverage --key javascript --value-file ./coverage/cobertura-coverage.xml
        env:
          DEEPSOURCE_DSN: ${{ secrets.BIFROST_DSN }}
