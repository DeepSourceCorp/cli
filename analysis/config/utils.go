package config

import (
	"log"
	"os"
	"path"
	"path/filepath"
	"strings"

	"github.com/bmatcuk/doublestar/v4"
)

/* Walks the `CODE_PATH` directory and returns all the files other than the ones present
 * in the .git folder for analysis in the form of a string array */
func readAllFiles(codePath string) ([]string, error) {
	fileCount := 0

	allFiles := make([]string, 0)

	err := filepath.Walk(codePath,
		func(path string, fileInfo os.FileInfo, err error) error {
			if err != nil {
				return err
			}
			fileCount++

			/* Check the following before appending to the list of files:
			 * - Should not be a directory
			 * - The walked file should not be present in .git folder
			 * TODO: Check why fileCount != 1 condition is used here */
			if !fileInfo.IsDir() || !strings.HasPrefix(path, filepath.Join(codePath, ".git")) && fileCount != 1 {
				allFiles = append(allFiles, path)
			}
			return nil
		})
	if err != nil {
		return allFiles, err
	}
	return allFiles, nil
}

// Returns the slice of files matching certain glob patterns
func (r *AnalysisRun) getMatchingFiles(patterns []string) ([]string, error) {
	matchedFiles := make([]string, 0)

	// Return all the files if no exclude_patterns are configured
	if len(patterns) == 0 {
		return matchedFiles, nil
	}

	for _, file := range r.AnalysisFiles {
		for i := range patterns {
			match, err := doublestar.Match(path.Join(r.CodePath, patterns[i]), file)
			if err != nil {
				log.Printf("Failed to match the file %s with pattern %s\n", file, patterns[i])
			}
			if match {
				matchedFiles = append(matchedFiles, file)
				break
			}
		}
	}
	return matchedFiles, nil
}

// Filters the analysis files and removes the files matching the exclude_patterns from them
// TODO: Improve the logic here
func (r *AnalysisRun) filterAnalysisFiles() {
	filteredFiles := []string{}
	for _, file := range r.AnalysisFiles {
		for _, excludedFile := range r.ExcludedFiles {
			if file == excludedFile {
				break
			}
			filteredFiles = append(filteredFiles, file)
		}
	}
	r.AnalysisFiles = filteredFiles
}
