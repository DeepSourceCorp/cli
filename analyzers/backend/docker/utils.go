package docker

import (
	"bufio"
	"encoding/json"
	"errors"
	"fmt"
	"io"
	"strings"
)

type ErrorLine struct {
	Error       string      `json:"error"`
	ErrorDetail ErrorDetail `json:"errorDetail"`
}

type ErrorDetail struct {
	Message string `json:"message"`
}

type DockerBuildResponse struct {
	Stream string `json:"stream"`
}

func CheckBuildResponse(rd io.Reader, showAllLogs bool) error {
	var lastLine []byte
	count := 0
	var currentStream string

	scanner := bufio.NewScanner(rd)
	for scanner.Scan() {
		lastLine = scanner.Bytes()
		d := &DockerBuildResponse{}
		err := json.Unmarshal(lastLine, d)
		if err != nil {
			return err
		}
		if d.Stream == "" || d.Stream == "\n" || strings.Contains(d.Stream, "--->") || strings.TrimSuffix(d.Stream, "\n") == currentStream {
			continue
		}
		currentStream = strings.TrimSuffix(d.Stream, "\n")
		if showAllLogs {
			fmt.Println(currentStream)
		}
		count++
	}

	errLine := &ErrorLine{}
	json.Unmarshal([]byte(lastLine), errLine)
	if errLine.Error != "" {
		return errors.New(errLine.Error)
	}
	return scanner.Err()
}


