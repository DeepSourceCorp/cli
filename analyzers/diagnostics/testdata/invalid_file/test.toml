name = "2do-Checker"
