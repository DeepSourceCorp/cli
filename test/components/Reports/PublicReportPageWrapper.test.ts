import { render } from '@testing-library/vue'
import { PublicReportPageWrapper } from '~/components/Reports'
import { cartesian, generateStringProps } from '~/test/utils'
import { ReportPageT } from '~/types/reportTypes'
import { ReportLevel } from '~/types/types'

test('renders PublicReportPageWrapper with all prop options', () => {
  const baseProps = {
    createdAt: 'Jul 25, 2022',
    ownerLogin: 'deepsourcelabs',
    repositoryList: [{ id: 'UmVwb3NpdG9yeTp6dmp2eXo=', name: 'asgard', isPrivate: true }]
  }

  const reportKeyOptions = generateStringProps(
    'reportKey',
    [ReportPageT.OWASP_TOP_10, ReportPageT.SANS_TOP_25, ReportPageT.DISTRIBUTION],
    false
  )

  const levelOptions = generateStringProps(
    'level',
    [ReportLevel.Owner, ReportLevel.Repository, ReportLevel.Enterprise],
    false
  )

  cartesian(reportKeyOptions, levelOptions).forEach((propCombination) => {
    const props = {
      ...baseProps,
      ...propCombination
    }

    const { html } = render(PublicReportPageWrapper, {
      props,
      stubs: {
        ZIcon: true,
        ZButton: true,
        ZMenuSection: true,
        ZMenuItem: true
      }
    })

    expect(html()).toMatchSnapshot(JSON.stringify(props))
  })
})
