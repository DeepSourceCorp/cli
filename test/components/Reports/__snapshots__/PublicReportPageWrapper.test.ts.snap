// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PublicReportPageWrapper with all prop options: {"createdAt":"Jul 25, 2022","ownerLogin":"deepsourcelabs","repositoryList":[{"id":"UmVwb3NpdG9yeTp6dmp2eXo=","name":"asgard","isPrivate":true}],"level":"ENTERPRISE","reportKey":"issue-distribution"} 1`] = `
"<main>
  <header class=\\"space-y-4 md:space-y-1 border-b border-ink-200 pb-8\\">
    <h1 class=\\"text-lg md:text-2xl font-semibold\\">Issue Distribution report</h1>
    <p class=\\"text-vanilla-400 tracking-wide text-xs md:text-sm font-medium\\">
      Created on Jul 25, 2022
    </p>
  </header>
  <section class=\\"space-y-14 mt-8 leading-8\\">
    <div class=\\"space-y-4 text-vanilla-400\\">
      <h1 id=\\"summary\\" class=\\"text-lg text-vanilla-100 font-semibold scroll-mt-8\\">Summary</h1>
      <p>
        DeepSource continuously scans deepsourcelabs’ source code to detect code health issues and
        provides guidance to correct them. This report provides the current and historical snapshot of code health issues in
        deepsourcelabs’ source code, limited to the repositories tracked on DeepSource. The code health
        issue detected by DeepSource includes, but is not limited to, these categories: bug risks, anti-patterns, style
        violations, performance issues, code coverage issues, and so on.
      </p>
    </div>
    <div class=\\"space-y-4 text-vanilla-400\\">
      <h1 id=\\"intended-use\\" class=\\"text-lg text-vanilla-100 font-semibold scroll-mt-8\\">Intended use of this report</h1>
      <p>
        This report can be used by the deepsourcelabs team to identify and remediate code health issues in their code base
        and make it more maintainable. The report can also be used by key internal and external stakeholders to understand
        the code quality and maintainability of deepsourcelabs.
      </p>
    </div>
    <div class=\\"space-y-4 text-vanilla-400\\">
      <h1 id=\\"continuous-code-health\\" class=\\"text-lg text-vanilla-100 font-semibold scroll-mt-8\\">
        Continuous Code Health
      </h1>
      <p>
        DeepSource integrates with deepsourcelabs’ software development
        workflows to provide developers and key stakeholders a way to identify and fix code health
        issues continuously and proactively. This report is dynamic. As
        deepsourcelabs’ team improves its source code security posture, the
        changes will be reflected in this report.
      </p>
    </div>
  </section>
</main>"
`;

exports[`renders PublicReportPageWrapper with all prop options: {"createdAt":"Jul 25, 2022","ownerLogin":"deepsourcelabs","repositoryList":[{"id":"UmVwb3NpdG9yeTp6dmp2eXo=","name":"asgard","isPrivate":true}],"level":"ENTERPRISE","reportKey":"owasp-top-10"} 1`] = `
"<main>
  <header class=\\"space-y-4 md:space-y-1 border-b border-ink-200 pb-8\\">
    <h1 class=\\"text-lg md:text-2xl font-semibold\\">OWASP Top 10 report</h1>
    <p class=\\"text-vanilla-400 tracking-wide text-xs md:text-sm font-medium\\">
      Created on Jul 25, 2022
    </p>
  </header>
  <section class=\\"space-y-14 mt-8 leading-8\\">
    <div class=\\"space-y-4 text-vanilla-400\\">
      <h1 id=\\"summary\\" class=\\"text-lg text-vanilla-100 font-semibold scroll-mt-8\\">Summary</h1>
      <p>
        The Open Web Application Security Project Foundation, or OWASP Foundation, is a non-profit
        foundation dedicated to improving software security. The foundation is community-driven and
        helps organizations build secure software through open-source tools, resources, and
        research. The OWASP Top 10 is a recommendation framework maintained by the OWASP Foundation
        since 2003. The list is a collection of security recommendations that helps organizations to
        write more secure code. It is updated periodically to accommodate for the advancements and
        developments in application security. More information on OWASP Top 10 can be found
        <a href=\\"https://deepsource.io/guides/owasp-top-10/\\" rel=\\"nofollow noopener noreferrer\\" target=\\"_blank\\" class=\\"text-juniper-500 font-medium\\">here</a>.
      </p>
      <p>
        DeepSource continuously scans deepsourcelabs’ source code continuously to detect violations
        of OWASP Top 10 recommendations and provides guidance to correct them. This report provides
        the current, as well as a historical snapshot of the presence of such vulnerabilities in
        deepsourcelabs’ source code, limited to the repositories, tracked on DeepSource.
      </p>
    </div>
    <div class=\\"space-y-4 text-vanilla-400\\">
      <h1 id=\\"intended-use\\" class=\\"text-lg text-vanilla-100 font-semibold scroll-mt-8\\">Intended use of this report</h1>
      <p>
        This report can be used by the deepsourcelabs team to identify and remediate violations of
        OWASP Top 10 recommendations in their source code. The report can also be used by key
        stakeholders within the organization and interested external stakeholders to understand the
        source code security posture of deepsourcelabs.
      </p>
    </div>
    <div class=\\"space-y-4 text-vanilla-400\\">
      <h1 id=\\"continuous-code-health\\" class=\\"text-lg text-vanilla-100 font-semibold scroll-mt-8\\">
        Continuous Code Health
      </h1>
      <p>
        DeepSource integrates with deepsourcelabs’ software development
        workflows to provide developers and key stakeholders a way to identify and fix code health
        issues continuously and proactively. This report is dynamic. As
        deepsourcelabs’ team improves its source code security posture, the
        changes will be reflected in this report.
      </p>
    </div>
  </section>
</main>"
`;

exports[`renders PublicReportPageWrapper with all prop options: {"createdAt":"Jul 25, 2022","ownerLogin":"deepsourcelabs","repositoryList":[{"id":"UmVwb3NpdG9yeTp6dmp2eXo=","name":"asgard","isPrivate":true}],"level":"ENTERPRISE","reportKey":"sans-top-25"} 1`] = `
"<main>
  <header class=\\"space-y-4 md:space-y-1 border-b border-ink-200 pb-8\\">
    <h1 class=\\"text-lg md:text-2xl font-semibold\\">SANS Top 25 report</h1>
    <p class=\\"text-vanilla-400 tracking-wide text-xs md:text-sm font-medium\\">
      Created on Jul 25, 2022
    </p>
  </header>
  <section class=\\"space-y-14 mt-8 leading-8\\">
    <div class=\\"space-y-4 text-vanilla-400\\">
      <h1 id=\\"summary\\" class=\\"text-lg text-vanilla-100 font-semibold scroll-mt-8\\">Summary</h1>
      <p>
        The SANS Institute was founded in 1989 as a cooperative for information security thought
        leadership. The CWE (Common Weakness Enumeration) / SANS Top 25 list is a well-known list
        of the most common and severe errors in software that can lead to serious security
        vulnerabilities. The list is used by organizations as a primer to ensure their source code
        is secure.
      </p>
      <p>
        DeepSource continuously scans deepsourcelabs’s source code continuously to detect violations
        of SANS Top 25 recommendations and provides guidance to correct them. This report provides
        the current, as well as a historical snapshot of the presence of such vulnerabilities in
        deepsourcelabs’s source code, limited to the repositories, tracked on DeepSource.
      </p>
    </div>
    <div class=\\"space-y-4 text-vanilla-400\\">
      <h1 id=\\"intended-use\\" class=\\"text-lg text-vanilla-100 font-semibold scroll-mt-8\\">Intended use of this report</h1>
      <p>
        This report can be used by the deepsourcelabs team to identify and remediate violations of
        SANS Top 25 recommendations in their source code. The report can also be used by key
        stakeholders within the organization and interested external stakeholders to understand the
        source code security posture of deepsourcelabs.
      </p>
    </div>
    <div class=\\"space-y-4 text-vanilla-400\\">
      <h1 id=\\"continuous-code-health\\" class=\\"text-lg text-vanilla-100 font-semibold scroll-mt-8\\">
        Continuous Code Health
      </h1>
      <p>
        DeepSource integrates with deepsourcelabs’ software development
        workflows to provide developers and key stakeholders a way to identify and fix code health
        issues continuously and proactively. This report is dynamic. As
        deepsourcelabs’ team improves its source code security posture, the
        changes will be reflected in this report.
      </p>
    </div>
  </section>
</main>"
`;

exports[`renders PublicReportPageWrapper with all prop options: {"createdAt":"Jul 25, 2022","ownerLogin":"deepsourcelabs","repositoryList":[{"id":"UmVwb3NpdG9yeTp6dmp2eXo=","name":"asgard","isPrivate":true}],"level":"OWNER","reportKey":"issue-distribution"} 1`] = `
"<main>
  <header class=\\"space-y-4 md:space-y-1 border-b border-ink-200 pb-8\\">
    <h1 class=\\"text-lg md:text-2xl font-semibold\\">Issue Distribution report</h1>
    <p class=\\"text-vanilla-400 tracking-wide text-xs md:text-sm font-medium\\">
      Created on Jul 25, 2022
    </p>
  </header>
  <section class=\\"space-y-14 mt-8 leading-8\\">
    <div class=\\"space-y-4 text-vanilla-400\\">
      <h1 id=\\"summary\\" class=\\"text-lg text-vanilla-100 font-semibold scroll-mt-8\\">Summary</h1>
      <p>
        DeepSource continuously scans deepsourcelabs’ source code to detect code health issues and
        provides guidance to correct them. This report provides the current and historical snapshot of code health issues in
        deepsourcelabs’ source code, limited to the repositories tracked on DeepSource. The code health
        issue detected by DeepSource includes, but is not limited to, these categories: bug risks, anti-patterns, style
        violations, performance issues, code coverage issues, and so on.
      </p>
    </div>
    <div class=\\"space-y-4 text-vanilla-400\\">
      <h1 id=\\"intended-use\\" class=\\"text-lg text-vanilla-100 font-semibold scroll-mt-8\\">Intended use of this report</h1>
      <p>
        This report can be used by the deepsourcelabs team to identify and remediate code health issues in their code base
        and make it more maintainable. The report can also be used by key internal and external stakeholders to understand
        the code quality and maintainability of deepsourcelabs.
      </p>
    </div>
    <div class=\\"space-y-4 text-vanilla-400\\">
      <h1 id=\\"continuous-code-health\\" class=\\"text-lg text-vanilla-100 font-semibold scroll-mt-8\\">
        Continuous Code Health
      </h1>
      <p>
        DeepSource integrates with deepsourcelabs’ software development
        workflows to provide developers and key stakeholders a way to identify and fix code health
        issues continuously and proactively. This report is dynamic. As
        deepsourcelabs’ team improves its source code security posture, the
        changes will be reflected in this report.
      </p>
    </div>
  </section>
</main>"
`;

exports[`renders PublicReportPageWrapper with all prop options: {"createdAt":"Jul 25, 2022","ownerLogin":"deepsourcelabs","repositoryList":[{"id":"UmVwb3NpdG9yeTp6dmp2eXo=","name":"asgard","isPrivate":true}],"level":"OWNER","reportKey":"owasp-top-10"} 1`] = `
"<main>
  <header class=\\"space-y-4 md:space-y-1 border-b border-ink-200 pb-8\\">
    <h1 class=\\"text-lg md:text-2xl font-semibold\\">OWASP Top 10 report</h1>
    <p class=\\"text-vanilla-400 tracking-wide text-xs md:text-sm font-medium\\">
      Created on Jul 25, 2022
    </p>
  </header>
  <section class=\\"space-y-14 mt-8 leading-8\\">
    <div class=\\"space-y-4 text-vanilla-400\\">
      <h1 id=\\"summary\\" class=\\"text-lg text-vanilla-100 font-semibold scroll-mt-8\\">Summary</h1>
      <p>
        The Open Web Application Security Project Foundation, or OWASP Foundation, is a non-profit
        foundation dedicated to improving software security. The foundation is community-driven and
        helps organizations build secure software through open-source tools, resources, and
        research. The OWASP Top 10 is a recommendation framework maintained by the OWASP Foundation
        since 2003. The list is a collection of security recommendations that helps organizations to
        write more secure code. It is updated periodically to accommodate for the advancements and
        developments in application security. More information on OWASP Top 10 can be found
        <a href=\\"https://deepsource.io/guides/owasp-top-10/\\" rel=\\"nofollow noopener noreferrer\\" target=\\"_blank\\" class=\\"text-juniper-500 font-medium\\">here</a>.
      </p>
      <p>
        DeepSource continuously scans deepsourcelabs’ source code continuously to detect violations
        of OWASP Top 10 recommendations and provides guidance to correct them. This report provides
        the current, as well as a historical snapshot of the presence of such vulnerabilities in
        deepsourcelabs’ source code, limited to the repositories, tracked on DeepSource.
      </p>
    </div>
    <div class=\\"space-y-4 text-vanilla-400\\">
      <h1 id=\\"intended-use\\" class=\\"text-lg text-vanilla-100 font-semibold scroll-mt-8\\">Intended use of this report</h1>
      <p>
        This report can be used by the deepsourcelabs team to identify and remediate violations of
        OWASP Top 10 recommendations in their source code. The report can also be used by key
        stakeholders within the organization and interested external stakeholders to understand the
        source code security posture of deepsourcelabs.
      </p>
    </div>
    <div class=\\"space-y-4 text-vanilla-400\\">
      <h1 id=\\"continuous-code-health\\" class=\\"text-lg text-vanilla-100 font-semibold scroll-mt-8\\">
        Continuous Code Health
      </h1>
      <p>
        DeepSource integrates with deepsourcelabs’ software development
        workflows to provide developers and key stakeholders a way to identify and fix code health
        issues continuously and proactively. This report is dynamic. As
        deepsourcelabs’ team improves its source code security posture, the
        changes will be reflected in this report.
      </p>
    </div>
  </section>
</main>"
`;

exports[`renders PublicReportPageWrapper with all prop options: {"createdAt":"Jul 25, 2022","ownerLogin":"deepsourcelabs","repositoryList":[{"id":"UmVwb3NpdG9yeTp6dmp2eXo=","name":"asgard","isPrivate":true}],"level":"OWNER","reportKey":"sans-top-25"} 1`] = `
"<main>
  <header class=\\"space-y-4 md:space-y-1 border-b border-ink-200 pb-8\\">
    <h1 class=\\"text-lg md:text-2xl font-semibold\\">SANS Top 25 report</h1>
    <p class=\\"text-vanilla-400 tracking-wide text-xs md:text-sm font-medium\\">
      Created on Jul 25, 2022
    </p>
  </header>
  <section class=\\"space-y-14 mt-8 leading-8\\">
    <div class=\\"space-y-4 text-vanilla-400\\">
      <h1 id=\\"summary\\" class=\\"text-lg text-vanilla-100 font-semibold scroll-mt-8\\">Summary</h1>
      <p>
        The SANS Institute was founded in 1989 as a cooperative for information security thought
        leadership. The CWE (Common Weakness Enumeration) / SANS Top 25 list is a well-known list
        of the most common and severe errors in software that can lead to serious security
        vulnerabilities. The list is used by organizations as a primer to ensure their source code
        is secure.
      </p>
      <p>
        DeepSource continuously scans deepsourcelabs’s source code continuously to detect violations
        of SANS Top 25 recommendations and provides guidance to correct them. This report provides
        the current, as well as a historical snapshot of the presence of such vulnerabilities in
        deepsourcelabs’s source code, limited to the repositories, tracked on DeepSource.
      </p>
    </div>
    <div class=\\"space-y-4 text-vanilla-400\\">
      <h1 id=\\"intended-use\\" class=\\"text-lg text-vanilla-100 font-semibold scroll-mt-8\\">Intended use of this report</h1>
      <p>
        This report can be used by the deepsourcelabs team to identify and remediate violations of
        SANS Top 25 recommendations in their source code. The report can also be used by key
        stakeholders within the organization and interested external stakeholders to understand the
        source code security posture of deepsourcelabs.
      </p>
    </div>
    <div class=\\"space-y-4 text-vanilla-400\\">
      <h1 id=\\"continuous-code-health\\" class=\\"text-lg text-vanilla-100 font-semibold scroll-mt-8\\">
        Continuous Code Health
      </h1>
      <p>
        DeepSource integrates with deepsourcelabs’ software development
        workflows to provide developers and key stakeholders a way to identify and fix code health
        issues continuously and proactively. This report is dynamic. As
        deepsourcelabs’ team improves its source code security posture, the
        changes will be reflected in this report.
      </p>
    </div>
  </section>
</main>"
`;

exports[`renders PublicReportPageWrapper with all prop options: {"createdAt":"Jul 25, 2022","ownerLogin":"deepsourcelabs","repositoryList":[{"id":"UmVwb3NpdG9yeTp6dmp2eXo=","name":"asgard","isPrivate":true}],"level":"REPOSITORY","reportKey":"issue-distribution"} 1`] = `
"<main>
  <header class=\\"space-y-4 md:space-y-1 border-b border-ink-200 pb-8\\">
    <h1 class=\\"text-lg md:text-2xl font-semibold\\">Issue Distribution report</h1>
    <p class=\\"text-vanilla-400 tracking-wide text-xs md:text-sm font-medium\\">
      Created on Jul 25, 2022
    </p>
  </header>
  <section class=\\"space-y-14 mt-8 leading-8\\">
    <div class=\\"space-y-4 text-vanilla-400\\">
      <h1 id=\\"summary\\" class=\\"text-lg text-vanilla-100 font-semibold scroll-mt-8\\">Summary</h1>
      <p>
        DeepSource continuously scans deepsourcelabs’ source code to detect code health issues and
        provides guidance to correct them. This report provides the current and historical snapshot of code health issues in
        deepsourcelabs’ source code, limited to the repositories tracked on DeepSource. The code health
        issue detected by DeepSource includes, but is not limited to, these categories: bug risks, anti-patterns, style
        violations, performance issues, code coverage issues, and so on.
      </p>
    </div>
    <div class=\\"space-y-4 text-vanilla-400\\">
      <h1 id=\\"intended-use\\" class=\\"text-lg text-vanilla-100 font-semibold scroll-mt-8\\">Intended use of this report</h1>
      <p>
        This report can be used by the deepsourcelabs team to identify and remediate code health issues in their code base
        and make it more maintainable. The report can also be used by key internal and external stakeholders to understand
        the code quality and maintainability of deepsourcelabs.
      </p>
    </div>
    <div class=\\"space-y-4 text-vanilla-400\\">
      <h1 id=\\"continuous-code-health\\" class=\\"text-lg text-vanilla-100 font-semibold scroll-mt-8\\">
        Continuous Code Health
      </h1>
      <p>
        DeepSource integrates with deepsourcelabs’ software development
        workflows to provide developers and key stakeholders a way to identify and fix code health
        issues continuously and proactively. This report is dynamic. As
        deepsourcelabs’ team improves its source code security posture, the
        changes will be reflected in this report.
      </p>
    </div>
  </section>
</main>"
`;

exports[`renders PublicReportPageWrapper with all prop options: {"createdAt":"Jul 25, 2022","ownerLogin":"deepsourcelabs","repositoryList":[{"id":"UmVwb3NpdG9yeTp6dmp2eXo=","name":"asgard","isPrivate":true}],"level":"REPOSITORY","reportKey":"owasp-top-10"} 1`] = `
"<main>
  <header class=\\"space-y-4 md:space-y-1 border-b border-ink-200 pb-8\\">
    <h1 class=\\"text-lg md:text-2xl font-semibold\\">OWASP Top 10 report</h1>
    <p class=\\"text-vanilla-400 tracking-wide text-xs md:text-sm font-medium\\">
      Created on Jul 25, 2022
    </p>
  </header>
  <section class=\\"space-y-14 mt-8 leading-8\\">
    <div class=\\"space-y-4 text-vanilla-400\\">
      <h1 id=\\"summary\\" class=\\"text-lg text-vanilla-100 font-semibold scroll-mt-8\\">Summary</h1>
      <p>
        The Open Web Application Security Project Foundation, or OWASP Foundation, is a non-profit
        foundation dedicated to improving software security. The foundation is community-driven and
        helps organizations build secure software through open-source tools, resources, and
        research. The OWASP Top 10 is a recommendation framework maintained by the OWASP Foundation
        since 2003. The list is a collection of security recommendations that helps organizations to
        write more secure code. It is updated periodically to accommodate for the advancements and
        developments in application security. More information on OWASP Top 10 can be found
        <a href=\\"https://deepsource.io/guides/owasp-top-10/\\" rel=\\"nofollow noopener noreferrer\\" target=\\"_blank\\" class=\\"text-juniper-500 font-medium\\">here</a>.
      </p>
      <p>
        DeepSource continuously scans deepsourcelabs’ source code continuously to detect violations
        of OWASP Top 10 recommendations and provides guidance to correct them. This report provides
        the current, as well as a historical snapshot of the presence of such vulnerabilities in
        deepsourcelabs’ source code, limited to the repositories, tracked on DeepSource.
      </p>
    </div>
    <div class=\\"space-y-4 text-vanilla-400\\">
      <h1 id=\\"intended-use\\" class=\\"text-lg text-vanilla-100 font-semibold scroll-mt-8\\">Intended use of this report</h1>
      <p>
        This report can be used by the deepsourcelabs team to identify and remediate violations of
        OWASP Top 10 recommendations in their source code. The report can also be used by key
        stakeholders within the organization and interested external stakeholders to understand the
        source code security posture of deepsourcelabs.
      </p>
    </div>
    <div class=\\"space-y-4 text-vanilla-400\\">
      <h1 id=\\"continuous-code-health\\" class=\\"text-lg text-vanilla-100 font-semibold scroll-mt-8\\">
        Continuous Code Health
      </h1>
      <p>
        DeepSource integrates with deepsourcelabs’ software development
        workflows to provide developers and key stakeholders a way to identify and fix code health
        issues continuously and proactively. This report is dynamic. As
        deepsourcelabs’ team improves its source code security posture, the
        changes will be reflected in this report.
      </p>
    </div>
  </section>
</main>"
`;

exports[`renders PublicReportPageWrapper with all prop options: {"createdAt":"Jul 25, 2022","ownerLogin":"deepsourcelabs","repositoryList":[{"id":"UmVwb3NpdG9yeTp6dmp2eXo=","name":"asgard","isPrivate":true}],"level":"REPOSITORY","reportKey":"sans-top-25"} 1`] = `
"<main>
  <header class=\\"space-y-4 md:space-y-1 border-b border-ink-200 pb-8\\">
    <h1 class=\\"text-lg md:text-2xl font-semibold\\">SANS Top 25 report</h1>
    <p class=\\"text-vanilla-400 tracking-wide text-xs md:text-sm font-medium\\">
      Created on Jul 25, 2022
    </p>
  </header>
  <section class=\\"space-y-14 mt-8 leading-8\\">
    <div class=\\"space-y-4 text-vanilla-400\\">
      <h1 id=\\"summary\\" class=\\"text-lg text-vanilla-100 font-semibold scroll-mt-8\\">Summary</h1>
      <p>
        The SANS Institute was founded in 1989 as a cooperative for information security thought
        leadership. The CWE (Common Weakness Enumeration) / SANS Top 25 list is a well-known list
        of the most common and severe errors in software that can lead to serious security
        vulnerabilities. The list is used by organizations as a primer to ensure their source code
        is secure.
      </p>
      <p>
        DeepSource continuously scans deepsourcelabs’s source code continuously to detect violations
        of SANS Top 25 recommendations and provides guidance to correct them. This report provides
        the current, as well as a historical snapshot of the presence of such vulnerabilities in
        deepsourcelabs’s source code, limited to the repositories, tracked on DeepSource.
      </p>
    </div>
    <div class=\\"space-y-4 text-vanilla-400\\">
      <h1 id=\\"intended-use\\" class=\\"text-lg text-vanilla-100 font-semibold scroll-mt-8\\">Intended use of this report</h1>
      <p>
        This report can be used by the deepsourcelabs team to identify and remediate violations of
        SANS Top 25 recommendations in their source code. The report can also be used by key
        stakeholders within the organization and interested external stakeholders to understand the
        source code security posture of deepsourcelabs.
      </p>
    </div>
    <div class=\\"space-y-4 text-vanilla-400\\">
      <h1 id=\\"continuous-code-health\\" class=\\"text-lg text-vanilla-100 font-semibold scroll-mt-8\\">
        Continuous Code Health
      </h1>
      <p>
        DeepSource integrates with deepsourcelabs’ software development
        workflows to provide developers and key stakeholders a way to identify and fix code health
        issues continuously and proactively. This report is dynamic. As
        deepsourcelabs’ team improves its source code security posture, the
        changes will be reflected in this report.
      </p>
    </div>
  </section>
</main>"
`;
